internal/error.go

```diff
@@ -419,3 +465,56 @@ func (b ErrorDetailsValues) Get(valuePtr ...interface{}) error {
 	}
 	return nil
 }
+
+// NewNonDeterminsticError constructs a new *NonDeterministicError.
+//
+//   - reason should be a documented NonDeterminsticError.Reason value
+//   - info is always required.  only a portion of it is used, but it is a convenient
+//     and currently always-available object.
+//   - history and decision may each be present or nil at any time
+func NewNonDeterminsticError(reason string, info *WorkflowInfo, history *shared.HistoryEvent, decision *shared.Decision) error {
+	var historyText string
+	if history != nil {
+		historyText = util.HistoryEventToString(history)
+	}
+	var decisionText string
+	if decision != nil {
+		decisionText = util.DecisionToString(decision)
+	}
+	return &NonDeterministicError{
+		Reason: reason,
+
+		WorkflowType: info.WorkflowType.Name,
+		WorkflowID:   info.WorkflowExecution.ID,
+		RunID:        info.WorkflowExecution.RunID,
+		DomainName:   info.Domain,
+
```

Comment:
Should we add the `TaskList:(Name:canary-task-queue)` object too since it was a part of the original error string?

Reply:
yea, seems reasonable